src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -33,57 +42,83 @@ public void onActorDeath(ActorDeath actor) {
         if (isEnabled() && plugin.getClient().getLocalPlayer() == actor.getActor()) {
             this.handleNotify();
         }
+        lastTarget = new WeakReference<>(null);
     }
 
-    private void handleNotify() {
-        Player localPlayer = plugin.getClient().getLocalPlayer();
-        Actor pker = null;
-        for (Player other : plugin.getClient().getPlayers()) {
-            if (other.getInteracting() == localPlayer) {
-                pker = other;
-                break;
-            }
+    public void onInteraction(InteractingChanged event) {
+        if (event.getSource() == plugin.getClient().getLocalPlayer()) {
+            lastTarget = new WeakReference<>(event.getTarget());
         }
-        ItemContainer inventory = plugin.getClient().getItemContainer(InventoryID.INVENTORY);
-        ItemContainer equipment = plugin.getClient().getItemContainer(InventoryID.EQUIPMENT);
+    }
 
-        assert inventory != null;
-        assert equipment != null;
-        List<Pair<Item, Long>> itemsByPrice = Stream.concat(Arrays.stream(inventory.getItems()), Arrays.stream(equipment.getItems()))
-            .map(item -> Pair.of(item, (long) (plugin.getItemManager().getItemPrice(item.getId())) * (long) (item.getQuantity())))
-            .sorted((item1, item2) -> Math.toIntExact(item2.getRight() - item1.getRight()))
-            .collect(Collectors.toList());
+    private void handleNotify() {
+        Player localPlayer = plugin.getClient().getLocalPlayer();
+        Actor pker = identifyPker();
 
-        int keepCount = 3;
-        if (localPlayer.getSkullIcon() != null) {
-            keepCount = 0;
-        }
-        if (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM)) {
-            keepCount += 1;
-        }
-        Long losePrice = itemsByPrice.stream()
+        List<Pair<Item, Long>> itemsByPrice = getPricedItems();
+        int keepCount = (localPlayer.getSkullIcon() == null ? 3 : 0) +
+            (plugin.getClient().isPrayerActive(Prayer.PROTECT_ITEM) ? 1 : 0);
```

Comment:
This change reduces readability, if the previous if-statements took up too much space it should be moved into a private function that performs similarly easy-to-read logic (like done with `getPricedItems`)